beefy-gadget/src/lib.rs

```diff
@@ -264,15 +268,17 @@ where
 		number == next_block_to_vote_on
 	}
 
-	fn sign_commitment(&self, id: &Id, commitment: &[u8]) -> Option<Signature> {
+	fn sign_commitment(&self, id: &Id, commitment: &[u8]) -> Result<Signature, Error> {
 		let sig = SyncCryptoStore::sign_with(&*self.key_store, KEY_TYPE, &id.to_public_crypto_pair(), &commitment)
-			.ok()
-			.flatten()?
+			.map_err(|e| Error::CannotSign(id.to_raw_vec(), e.to_string()))?
```

Comment:
I suspect you don't want to make the `Error` generic, hence vec conversion right?

Reply:
Error type is now generic of `Id`.